### ISSUE TITLE
react/require-render-return shouldn't warn if render is defined using (stage 1) class property syntax

### ISSUE THREAD

**Issue by jimbolla** [yannickcr/eslint-plugin-react#564]
`react/require-render-return` will error if a class is defined with this syntax:

```
export default class UserList extends React.Component {
  render = () => (
    <div/>
  );
}
```

We use this syntax to make our class function declarations consistent between render and event handler functions.

http://babeljs.io/docs/plugins/transform-class-properties/

**Comment by ljharb** [comment]
Instance properties are not yet standard - and doing this (instead of `render() {`) is actually much slower, and I think is something that should be discouraged.

**Comment by billyjanitsch** [comment]
@ljharb why would it be slower? Doesn't React already bind all standard component methods to the instance in the constructor anyway?

**Comment by ljharb** [comment]
@billyjanitsch only in `React.createClass`, not in `class`-based components.

**Comment by ljharb** [comment]
@yannickcr this change is breaking something in eslint-config-airbnb
```sh
Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at getPropertyName ($PROJECT/node_modules/eslint-plugin-react/lib/rules/require-render-return.js:54:20)
    at Object.ArrowFunctionExpression ($PROJECT/node_modules/eslint-plugin-react/lib/rules/require-render-return.js:93:40)
    at EventEmitter.updatedRuleInstructions.(anonymous function) ($PROJECT/node_modules/eslint-plugin-react/lib/util/Components.js:457:75)
    at emitOne (events.js:101:20)
    at EventEmitter.emit (events.js:188:7)
    at NodeEventGenerator.enterNode ($PROJECT/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode ($PROJECT/node_modules/eslint/lib/code-path-analysis/code-path-analynpm ERR! Test failed.  See above for more details.
```

**Comment by yannickcr** [comment]
@ljharb Ho. Do you know what code makes it crash ?

**Comment by vdh** [comment]
@yannickcr It looks similar to this issue in eslint-plugin-import: https://github.com/benmosher/eslint-plugin-import/issues/317

**Comment by ljharb** [comment]
I'm afraid I don't - it doesn't tell me which.

**Comment by ljharb** [comment]
fwiw I'm not using any proposed ES syntax.

**Comment by yannickcr** [comment]
Ok, the following pattern trigger the crash:
```js
var Hello = () => (
  <div></div>
);
```
I'll do a quick fix.
